app/src/main/java/com/github/se/gomeet/ui/authscreens/register/RegisterNameCountryPhone.kt

```diff
@@ -37,12 +37,91 @@ import androidx.compose.ui.Modifier
 import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.platform.LocalConfiguration
 import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.text.buildAnnotatedString
 import androidx.compose.ui.text.input.ImeAction
 import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.OffsetMapping
+import androidx.compose.ui.text.input.TransformedText
+import androidx.compose.ui.text.input.VisualTransformation
 import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.unit.dp
 import com.github.se.gomeet.model.authentication.getCountries
 
+/**
+ * A visual transformation for formatting phone numbers in the international format.
+ *
+ * This class formats phone numbers to insert spaces at appropriate positions. For example, a phone
+ * number like `+41227676111` will be transformed to `+41 22 767 6111`.
+ *
+ * This transformation is designed to handle the following format:
+ * - The first 3 characters, including the `+` sign and country code, are grouped together.
+ * - A space follows the country code.
+ * - The next 2 characters form the area code.
+ * - A space follows the area code.
+ * - The next 3 characters form the prefix.
+ * - A space follows the prefix.
+ * - The remaining characters form the subscriber number.
+ *
+ * Example usage:
+ * ```
+ * TextField(
+ *     value = phoneNumber,
+ *     onValueChange = { phoneNumber = it },
+ *     label = { Text("Phone Number") },
+ *     singleLine = true,
+ *     colors = textFieldColors,
+ *     keyboardOptions = KeyboardOptions.Default.copy(
+ *         imeAction = ImeAction.Done,
+ *         keyboardType = KeyboardType.Phone
+ *     ),
+ *     visualTransformation = PhoneNumberVisualTransformation(),
+ *     modifier = Modifier.fillMaxWidth()
+ * )
+ * ```
+ *
+ * @see VisualTransformation
+ */
+class PhoneNumberVisualTransformation : VisualTransformation {
+  override fun filter(text: AnnotatedString): TransformedText {
+    val trimmed = if (text.text.length >= 15) text.text.substring(0..14) else text.text
+    val formatted = buildAnnotatedString {
+      append(trimmed.take(3)) // +41
+      if (trimmed.length > 3) append(" ")
```

Comment:
For the coaches: I am letting these " " magic strings through because I don't believe that they are a problem, since these are just spaces being appended for the format of the phone number :)